ludwig/models/llm.py

```diff
@@ -452,19 +621,39 @@ def realign_target_and_prediction_tensors(
     if target_length == prediction_length:
         return targets, predictions
 
+    if pad_direction not in {"left", "right"}:
+        raise ValueError(f'pad_direction must be either "left" or "right". Got {pad_direction}.')
+
+    if not pad_value:
+        pad_value = tokenizer.pad_token_id if tokenizer.pad_token_id is not None else tokenizer.eos_token_id
+
     # Align target and prediction tensors for text to text metric computation
     if target_length > prediction_length:
         # Pad the predictions.
         zeros_to_add = target_length - prediction_length
-        predictions[of_name][PREDICTIONS] = F.pad(predictions[of_name][PREDICTIONS], (0, zeros_to_add))
+
+        if pad_direction == "right":
+            predictions[of_name][PREDICTIONS] = F.pad(
+                predictions[of_name][PREDICTIONS], (0, zeros_to_add), value=pad_value
+            )
+            predictions[of_name][PROBABILITIES] = F.pad(predictions[of_name][PROBABILITIES], (0, 0, 0, zeros_to_add))
```

Comment:
Can you add a test for this function? I can't easily tell if this actually works